### DISCUSSION TITLE
Converting .webp to .jpg

### DISCUSSION THREAD

**Discussion by JackyYu510yt** [ImageMagick/ImageMagick#7838] [Help]
### ImageMagick version

7.1.1-41

### Operating system, version and so on

WIn 10

### Description

I'm trying to download some manhwa using HakuNeko, but for some of the manhwa, they're downloading as .webp despite it being set to download preferably as a .jpg.
To combat this, I was told to use a post command. I went to the FAQ and found this
mkdir "%PATH%_conv" 2>nul & convert "%PATH%\*.webp" -scene 1 "%PATH%_conv\%03d.png"
I ended up replacing the .png with a .jpg 

It was working initially, but it just stopped working for me recently. No idea why
It also wasn't that great since it would create another folder with the same name, just with a suffix. It made it annoying to select.
Ex) Chapter 1 
     Chapter 1_conv
     Chapter 2 

Instead, would it be possible to just replace the contents of the original Chapter 1 file with the .jpg files that would've been in the _conv folder?
I was given this post command by somebody else, but it didn't seem to work.
convert "%PATH%*.webp" -scene 1 "%PATH%%03d.jpg

Please advise, and thank you.

**Comment by snibgo** [comment]
`convert` is deprecated. Use `magick` instead, so your command would be:
```
magick "%PATH%*.webp" -scene 1 "%PATH%_conv%03d.png"
```
That command would read all the images from all the webp files into memory. Then it would write those images as png files. If your total webp images are larger than your available memory, there may be problems, which might explain "it just stopped working for me recently".

Instead, I suggest `magick mogrify`, like this:
```
magick mogrify -path any/where/you/want/ -format PNG "%PATH%*.webp"
```
The `-path` option is the required output directory. It must already exist. The output files will be named after the input files, but with PNG extensions.

**Reply by JackyYu510yt** [reply]
Hey thanks for the fast response. Sorry I'm a complete noob when it comes to this stuff so I couldn't get it to work. I tried the first command you gave me, but it didn't seem to work. Not sure if I'm doing something wrong or if the program doesn't support the command.
For the second command, I'm a bit confused. Would I have to change the "-path any/where/you/want/" part to the folder I want it to go in? If so, how would I do that? Would it be 

> magick mogrify -path D:\Videos\Exports from HakuNeko\Thumbs Up, Level Up -format PNG "%PATH%*.webp"

**Comment by snibgo** [comment]
The command interpreter splits the commands into "words", assuming that words are separated by spaces, so it would split at the space in "Thumbs Up" etc. To avoid this, use quotes around the path:
```
magick mogrify -path "D:\Videos\Exports from HakuNeko\Thumbs Up, Level Up" -format PNG "%PATH%*.webp"
```
Personally, I don't use spaces or commas in directory or file names. They just cause problems.

**Reply by JackyYu510yt** [reply]
I tried copy and pasting the post command above and I'm still getting no results unfortunately. I'm not sure if it's the post command or if its just the program. Any other revisions I could try? The program im trying to use is HakuNeko if that matters.

**Comment by snibgo** [comment]
> ... and I'm still getting no results unfortunately.

What happens? Are there any warning or error messages? What is the value of %PATH%? (Incidentally, that is a reserved environment variable in Windows, and should not be used in arguments to ImageMagick.) Do you have any input files? Does the output directory exist? Does the command make any output files?

**Reply by JackyYu510yt** [reply]
The program just runs as normal and download as .webp.
Im assuming nothing is happening. 
What do you mean by the value of %PATH%? Is that the folder path? If so, I just used 

> "D:\Videos\Exports from HakuNeko\Thumbs Up, Level Up"

There was no input files and the outfit directory is just the normal Chapter 1 output.
https://i.imgur.com/LBlImOI.png

When I run it with the Post Command provided in the HakuNeko FAQ, it creates another folder, but has nothing inside. Previously it did, but now when I run it, it doesn't so it might be the memory issue you mentioned.
mkdir "%PATH%_conv" 2>nul & convert "%PATH%*.webp" -scene 1 "%PATH%_conv%03d.png"
https://i.imgur.com/KAeHVX0.png
https://i.imgur.com/Sq7bBF8.gif

**Comment by snibgo** [comment]
I know nothing about HakuNeko. My advice is about running ImageMagick commands at the command line, not within HakuNeko.
